File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0, r = numbers.size() - 1;
        while (l < r) {
            auto sum = numbers[l] + numbers[r];
            if (sum > target)
                r--;
            else if (sum < target)
                l++;
            else
                return {++l, ++r};
        }
        return {};
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
    s, e := 0, len(numbers) - 1

    for s < e {
        if numbers[s] + numbers[e] > target {
            e--
        } else if numbers[s] + numbers[e] < target {
            s++
        } else {
            return []int{s+1, e+1}
        }
    }

    return []int{-1, -1}
}